### PR TITLE
Re-pr of #70522 "Space Dragon Update: Up Close and Personal"

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

This PR is a re-pr of ##70522 , with some tweaks: 

Notably:

- Wavespeak is not a say override, but instead uses a mindlink. Meaning carp and space dragons can still talk verbally, but they can also use telepathy to talk to all carp and the dragon. 
    - I would refactor Mind Linker a bit further to be a full datum rather than a component but that's for another time. 
- Removed the gravity aura component in favor of using the existing forced gravity proximity monitor. 
    - Also fixed a bug involving that. Lol.  
- Minor refactoring around the place.
- Reduced the volume on a lot of space dragon sounds. 
- Edited the roundend report for Space Dragons to collate all entries into one per player. 

image

## Why It's Good For The Game

Space dragon still plays pretty "play lame win game" right now, the optimal strategy for them is to find the cheesiest spot for a portal and spam their stun / fire breath to make it unreachable. 

I was a fan of the original PR so I updated it and brought it back. 

## Changelog

:cl: IndieanaJones, Melbert
balance: Space Dragon can no longer choose its rift locations freely, and instead is given 5 pre-determined locations to pick from instead
balance: Space Dragon itself has been buffed in order to support a more confrontational playstyle, however its wing gust now requires a line of sight to targets in order to affect them.
balance: Player Space Carp from rifts now have buffed health, but reduced object damage values. They also gain a temporary speed boost when hit by Space Dragon's fire breath instead of taking damage.
balance: Carp rift spawn times have been reduced, the healing AOE is now a 3x3 instead of a 1x1, and apply normal gravity in a large radius around them
balance: Space Dragon and rift carps now communicate on a private mind link channel via action button similar to Raw Prophets and Slimepeople. 
fix: Fixed Gravity Generator forced gravity not applying. 
fix: Intern Announcer will no longer replace Space Dragon announcements.
qol: The roundend report for space dragons now collates all players who played a carp into one entry, rather than one per carp spawned. 
qol: Space Dragon sounds are much less ear piercingly loud. 
/:cl:

### CODE REVIEW

**Comment by nukularpower** [comment]
Not a fan of the forced locations at all as it will be very likely that all 5 are terrible.  It is already extremely rare to see a dragon get past 1 portal, though it does happen.    Wing Gust does not come out fast enough, hopefully its been sped up a bit so its not so easy to dodge by people that know what dragons do.   

The specific buffs arent listed here but would love it if there was some % chance of getting a magicarp or shark instead of just normal carps.   Maybe starting very small and getting somewhat higher as the portal got closer to completion or something like that.   

Willing to bet that we will be seeing a lot more failed dragon runs with this as is, and as I said, the vast majority of dragons (that I have observed, at any rate) do not get past portal 1 even with the "cheese strat" mentioned.

**Comment by moocowswag** [comment]
I know its not thematic but if space dragon is going to be more confrontational it is going to need to either be a bullet hell boss or have a way to slow down the stream of people coming to kill its portal.

**Comment by Mechar418** [comment]
Please, remove forced locations. Picking the right spot is the part of dragon's defense, that makes it more versatile.

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 1]
```diff
@@ -366,20 +393,22 @@
...
+		visible_message(span_boldwarning("[candidate] is knocked back by the gust!"))
+		to_chat(candidate, span_userdanger("You're knocked back by the gust!"))
+		var/dir_to_target = get_dir(get_turf(src), get_turf(candidate))
+		var/throwtarget = get_edge_target_turf(target, dir_to_target)
+		candidate.safe_throw_at(throwtarget, 10, 1, src)
+		candidate.Paralyze(50)
 	addtimer(CALLBACK(src, PROC_REF(reset_status)), 4 + ((tiredness * tiredness_mult) / 10))
 	tiredness = tiredness + (gust_tiredness * tiredness_mult)
 
+/mob/living/simple_animal/hostile/space_dragon/spawn_with_antag
```
Comment (Mothblocks) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:408]: What's the point of this? Is this a debug thing?

Comment (Mothblocks) [reply code/modules/mob/living/simple_animal/hostile/space_dragon.dm:408]: Why can't we have this be part of the normal mind_initialize?

**Comment by MrMelbert** [comment]
> Please, remove forced locations. Picking the right spot is the part of dragon's defense, that makes it more versatile.

But that's the whole point of the PR

**Comment by DATA-xPUNGED** [comment]
I think that specific locations might be fine with the numerous buffs, but we need to playtest it to be sure

**Comment by IndieanaJones** [comment]
I wasn't expecting anyone to pick this up before I could get around to fiddling with it again, but I consider it quite the pleasant surprise. The only thing I intended on maybe adding to my original PR outside of the alternate mind chat thing was giving SD a method to give nearby carps damage resistance, maybe like a roar that buffs them or something like that. Something to encourage carps to stay and fight with the dragon, because as it is right now, 2 laser shots will kill them. They also might be mitigated by the increased spawns as well though, so who knows.

**Comment by jlsnow301** [comment]
I actually wish wave speak did overwrite say, it'd be more in line with blob and I think it's a bit anticlimactic that people just talk to the fish. I saw the dragon get married in the chapel the other day idk. You have the chance to prevent these horrors

**Comment by MrMelbert** [comment]
> I saw the dragon get married in the chapel the other day idk. You have the chance to prevent these horrors

But that's fun

**Comment by jlsnow301** [comment]
> But that's fun

a man lies with a dragon and you're laughing

**Comment by nukularpower** [comment]
Seeing this in action seems to have proven my predictions true.  I really don't get why carp needed nerfed even more after the speed nerf when they got rift (a very poor trade imo) - now literally anything 1 shots them.   Give dragon a real bit of fun by adding magicarps and/or sharks as rare spawns imo

**Pull request merged**